src/js_types.hpp

```diff
@@ -352,12 +350,31 @@ REALM_JS_INLINE void set_internal(const typename T::Object &object, typename Cla
 template<typename T>
 inline bool Value<T>::is_valid_for_property(ContextType context, const ValueType &value, const Property& prop)
 {
-    if (prop.is_nullable && (is_null(context, value) || is_undefined(context, value))) {
+    if (is_nullable(prop.type) && (is_null(context, value) || is_undefined(context, value))) {
         return true;
     }
 
-    using PropertyType = realm::PropertyType;
-    switch (prop.type) {
+    using realm::PropertyType;
+    if (realm::is_array(prop.type)) {
+        if (prop.type != PropertyType::Object) {
+            return false;
+        }
+
+        // FIXME: Do we need to validate the types of the contained objects?
```

Comment:
How much does core/object store check?

Reply:
All of this validation is just to provide better error messages; none of it is semantically required.

Reply:
Let's leave it as an exercise for future contributors :smile: